axelor-bank-payment/src/main/java/com/axelor/apps/bankpayment/service/batch/BatchDirectDebitPaymentSchedule.java

```diff
@@ -94,15 +94,20 @@ protected void processPaymentScheduleLines(int paymentScheduleType) {
       Preconditions.checkNotNull(
           companyBankDetails.getCurrency(),
           I18n.get("Currency in company bank details is missing."));
-      Preconditions.checkArgument(
-          new File(accountingBatch.getPaymentMode().getBankOrderExportFolderPath()).exists(),
-          String.format(
-              I18n.get("Bank order export folder does not exist: %s"),
-              accountingBatch.getPaymentMode().getBankOrderExportFolderPath()));
 
-      BankPaymentConfigService bankPaymentConfigService = Beans.get(BankPaymentConfigService.class);
+      String bankOrderExportPath = accountingBatch.getPaymentMode().getBankOrderExportFolderPath();
 
       try {
+        String dataExpotDir = appBaseService.getDataExportDir();
+        Preconditions.checkArgument(
+            new File(dataExpotDir + File.separator + bankOrderExportPath).exists(),
+            String.format(
+                I18n.get("Bank order export folder does not exist: %s"),
+                dataExpotDir + File.separator + bankOrderExportPath));
```

Comment:
Better to keep Preconditions.checkArgument after catch block

Reply:
we can't move this condition to after catch block because it uses dataExpotDir variable and it must have to stay inside the try block as dataExpotDir is not accessible outside of try block.